canton/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/platform/apiserver/services/admin/ApiPackageManagementService.scala

```diff
@@ -133,6 +134,13 @@ private[apiserver] final class ApiPackageManagementService private (
           Future { () }
         } else
           packageUpgradeValidator.validateUpgrade(decodedDar.main)
+      _ <-
+        if (dryRun)
+          Future.failed(
+            Validation.DryRun.Error(decodedDar.main._1)
+              .asGrpcError)
+        else
+          Future{()}
```

Comment:
Feels a little odd to have this logic in the decode/validate function? should it not be between this validation and the code that "uploads"/sends the topology transactions